cli/module_generate/_templates/go/tmpl-Makefile

```diff
@@ -1,32 +1,41 @@
-BIN_OUTPUT_PATH = bin
-TOOL_BIN = bin/gotools/$(shell uname -s)-$(shell uname -m)
-UNAME_S ?= $(shell uname -s)
-GOPATH = $(HOME)/go/bin
-export PATH := ${PATH}:$(GOPATH)
 
-build: format update-rdk
-	rm -f $(BIN_OUTPUT_PATH)/{{ .ModuleName}}
-	go build $(LDFLAGS) -o $(BIN_OUTPUT_PATH)/{{ .ModuleName}} cmd/module/main.go
+GO_BUILD_ENV :=
+GO_BUILD_FLAGS :=
+MODULE_BINARY := bin/{{ .ModuleName }}
 
-module.tar.gz: build
-	rm -f $(BIN_OUTPUT_PATH)/module.tar.gz
-	tar czf $(BIN_OUTPUT_PATH)/module.tar.gz $(BIN_OUTPUT_PATH)/{{ .ModuleName}}
+ifeq ($(VIAM_TARGET_OS), windows)
+	GO_BUILD_ENV += GOOS=windows GOARCH=amd64
+	GO_BUILD_FLAGS := -tags no_cgo	
+	MODULE_BINARY = bin/{{ .ModuleName }}.exe
+endif
 
-setup:
-	if [ "$(UNAME_S)" = "Linux" ]; then \
-		sudo apt-get install -y apt-utils coreutils tar libnlopt-dev libjpeg-dev pkg-config; \
-	fi
-	# remove unused imports
-	go install golang.org/x/tools/cmd/goimports@latest
-	find . -name '*.go' -exec $(GOPATH)/goimports -w {} +
+$(MODULE_BINARY): Makefile go.mod *.go cmd/module/*.go 
+	$(GO_BUILD_ENV) go build $(GO_BUILD_FLAGS) -o $(MODULE_BINARY) cmd/module/main.go
 
+lint:
+	gofmt -s -w .
 
-clean:
-	rm -rf $(BIN_OUTPUT_PATH)/{{ .ModuleName}} $(BIN_OUTPUT_PATH)/module.tar.gz {{ .ModuleName}}
+update:
+	go get go.viam.com/rdk@latest
+	go mod tidy
 
-format:
-	gofmt -w -s .
+test:
+	go test ./...
 
-update-rdk:
-	go get go.viam.com/rdk@latest
+module.tar.gz: meta.json $(MODULE_BINARY)
+ifeq ($(VIAM_TARGET_OS), windows)
+	jq '.entrypoint = "./bin/{{ .ModuleName }}.exe"' meta.json > temp.json && mv temp.json meta.json
+else
+	strip $(MODULE_BINARY)
+endif
+	tar czf $@ meta.json $(MODULE_BINARY)
+ifeq ($(VIAM_TARGET_OS), windows)
+	git checkout meta.json
+endif
+
+module: test module.tar.gz
+
+all: test bin/viamkasa module 
```

Comment:
My ignorance: what is viamkasa? No need to explain if Ethan/Nicole recognize it.

Reply:
will fix